std/haxe/macro/ExprTools.hx

```diff
@@ -128,6 +128,11 @@ class ExprTools {
 				}
 				if (edef != null && edef.expr != null)
 					f(edef);
+			case EFormat(parts):
+				for (part in parts) switch part.kind {
+					case FRaw(_) | FIdent(_):
```

Comment:
This was my concern about having separate `FIdent` variant instead of having it all in `FExpr`: when itering/mapping over an expression, special handling is needed for format strings, if one wants to e.g. change all identifiers. OTOH this of course preserves more info about the source, so it may be useful in some cases. I don't have a strong opinion here.